File: teenpatti-backend/app/deck.py
# builds 52-card deck and shuffles it

import random

SUITS = ["spades", "hearts", "diamonds", "clubs"]
RANKS = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A"]


def build_deck():
    return [{"rank": rank, "suit": suit, "id": f"{rank}-{suit}"} for suit in SUITS for rank in RANKS]


def shuffle(deck):
    copy = list(deck)
    random.shuffle(copy)
    return copy

File: teenpatti-frontend/src/utils/deck.js
export const SUITS = ['spades', 'hearts', 'diamonds', 'clubs'];
export const RANKS = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K', 'A'];

export const SUIT_SYMBOL = {
  spades: '♠',
  hearts: '♥',
  diamonds: '♦',
  clubs: '♣',
};

export const SUIT_COLOR = {
  spades: 'dark',
  clubs: 'dark',
  hearts: 'red',
  diamonds: 'red',
};

export function buildDeck() {
  const deck = [];
  for (const suit of SUITS) {
    for (const rank of RANKS) {
      deck.push({ rank, suit, id: `${rank}-${suit}` });
    }
  }
  return deck;
}

export function shuffle(deck) {
  const copy = [...deck];
  for (let i = copy.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [copy[i], copy[j]] = [copy[j], copy[i]];
  }
  return copy;
}
